src/ControlzEx/Controls/TabControlEx.cs

```diff
@@ -0,0 +1,354 @@
+        /// </summary>
+        protected override void OnItemsChanged(NotifyCollectionChangedEventArgs e)
+        {
+            base.OnItemsChanged(e);
+
+            if (this.itemsHolder == null)
+            {
+                return;
+            }
+
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Reset:
+                    this.RefreshItemsHolder();
+                    break;
+
+                case NotifyCollectionChangedAction.Add:
+                case NotifyCollectionChangedAction.Remove:
+                    if (e.OldItems != null)
+                    {
+                        foreach (var item in e.OldItems)
+                        {
+                            var contentPresenter = this.FindChildContentPresenter(item);
+
+                            if (contentPresenter != null)
+                            {
+                                this.itemsHolder.Children.Remove(contentPresenter);
+                            }
+                        }
+                    }
+
+                    // don't do anything with new items because we don't want to
+                    // create visuals that aren't being shown yet
+                    this.UpdateSelectedItem();
+                    break;
+
+                case NotifyCollectionChangedAction.Replace:
+                    throw new NotImplementedException("Replace not implemented yet");
+            }
+        }
+
+        /// <summary>
+        /// Update the visible child in the ItemsHolder.
+        /// </summary>
+        protected override void OnSelectionChanged(SelectionChangedEventArgs e)
+        {
+            base.OnSelectionChanged(e);
+
+            this.UpdateSelectedItem();
+        }
+
+        /// <inheritdoc />
+        protected override void OnKeyDown(KeyEventArgs e)
+        {
+            // We need this to prevent the base class to always accept CTRL + TAB navigation regardless of which keyboard navigation mode is set for this control
+            if (e.Key == Key.Tab
+                && (e.KeyboardDevice.Modifiers & ModifierKeys.Control) == ModifierKeys.Control)
+            {
+                var controlTabNavigation = KeyboardNavigation.GetControlTabNavigation(this);
+
+                if (controlTabNavigation != KeyboardNavigationMode.Cycle
+                    && controlTabNavigation != KeyboardNavigationMode.Continue)
+                {
+                    return;
+                }
+            }
+
+            base.OnKeyDown(e);
+        }
+
+        /// <inheritdoc />
+        protected override AutomationPeer OnCreateAutomationPeer()
+        {
+            var automationPeer = new TabControlExAutomationPeer(this);
+            return automationPeer;
+        }
+
+        /// <summary>
+        /// Copied from <see cref="TabControl"/>. wish it were protected in that class instead of private.
+        /// </summary>
+        protected object GetSelectedItem()
```

Comment:
@batzen Is there anything against making this method public?

Reply:
Can't think of a reason to not make this public. Will change it tomorrow as i won't turn on my computer today 😉